workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectCreation/SpatialOSEntity.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Improbable.Worker;
+using Unity.Entities;
+
+namespace Improbable.Gdk.Core.GameObjectRepresentation
+{
+    /// <summary>
+    ///     Wrapper around an ECS Entity instance representing a SpatialOS Entity used for interacting with the the
+    ///     entity without touching ECS code like EntityManagers.
+    /// </summary>
+    public struct SpatialOSEntity
+    {
+        private readonly EntityManager entityManager;
+        private readonly Entity entity;
+        public readonly EntityId SpatialEntityId;
+
+        internal SpatialOSEntity(Entity entity, EntityManager entityManager)
+        {
+            this.entity = entity;
+            this.entityManager = entityManager;
+            SpatialEntityId = entityManager.GetComponentData<SpatialEntityId>(entity).EntityId;
+            UnityEntityIndex = entity.Index;
+        }
+
+        public bool HasComponent<T>() where T : struct, ISpatialComponentData, IComponentData
+        {
+            return entityManager.HasComponent<T>(entity);
+        }
+
+        public T GetComponent<T>() where T : struct, ISpatialComponentData, IComponentData
+        {
+            return entityManager.GetComponentData<T>(entity);
+        }
+
+        public int GetUnityEntityIndex()
+        {
+            return entity.Index;
```

Comment:
coming back to the comment before: why not make the entity public instead of introducing a getter that does not uniquely identify the entity?

Reply:
See above - this class is meant to hide ECS details and give a nice straightforward API to access SpatialOS entity component data. It's true that providing the index is a half-hearted step in the other direction though, we should probably either not have it at all or have a getter for the ECS entity which someone might want to use for hybrid workflows I guess...

Reply:
either expose the entity or not at all. The index does not give enough information about the entity

Reply:
Discussed with Firtina, I'll remove it